pkgs/binary-merkle-tree-halo2/src/binary_merkle_tree.rs

```diff
@@ -0,0 +1,264 @@
+#![warn(dead_code)]
+use anyhow::{Context, Ok, Result};
+use halo2_base::utils::BigPrimeField;
+use pse_poseidon::Poseidon;
+use rayon::{iter::ParallelIterator, slice::ParallelSlice};
+use serde::{Deserialize, Serialize};
+
+pub struct BinaryMerkleTree<'a, F, const T: usize, const RATE: usize, const ARITY: usize>
+where
+    F: BigPrimeField,
+{
+    pub root: F,
+    leaves: Vec<F>,
+    poseidon: &'a mut Poseidon<F, T, RATE>,
+    is_tree_ready: bool,
```

Comment:
what about a shorter
```suggestion
    ready: bool,
```